frontend/app/assets/javascripts/spree/frontend/checkout/coupon-code.js.coffee

```diff
@@ -0,0 +1,39 @@
+Spree.onCouponCodeApply = (e) ->
+  coupon_code_field = $('#order_coupon_code')
+  coupon_code = $.trim(coupon_code_field.val())
+  return if coupon_code == ''
+
+  if $('#coupon_status').length == 0
```

Comment:
I agree. Took it from the old script, sort of _backwards view compatible_, but I guess given that the view needs a button now, I can get rid of this.